packages/fdc3-agent-proxy/test/support/responses/RegisterListeners.ts

```diff
@@ -1,30 +1,53 @@
+import {
+  AddContextListenerRequest,
+  AddContextListenerResponse,
+  AddEventListenerRequest,
+  AddEventListenerResponse,
+  AddIntentListenerRequest,
+  AddIntentListenerResponse,
+  AppRequestMessage,
+  PrivateChannelAddEventListenerRequest,
+  PrivateChannelAddEventListenerResponse,
+} from '@kite9/fdc3-schema/generated/api/BrowserTypes';
 import { AutomaticResponse, TestMessaging } from '../TestMessaging';
 import { createResponseMeta } from './support';
 import { v4 as uuidv4 } from 'uuid';
 
+type requests =
+  | AddContextListenerRequest
+  | AddIntentListenerRequest
+  | AddEventListenerRequest
+  | PrivateChannelAddEventListenerRequest;
+type responses =
+  | AddContextListenerResponse
+  | AddIntentListenerResponse
+  | AddEventListenerResponse
+  | PrivateChannelAddEventListenerResponse;
+
 export class RegisterListeners implements AutomaticResponse {
   filter(t: string) {
     return (
       t == 'addContextListenerRequest' ||
       t == 'addIntentListenerRequest' ||
-      t == 'privateChannelAddContextListenerRequest' ||
+      t == 'addEventListenerRequest' ||
       t == 'privateChannelAddEventListenerRequest'
     );
   }
 
-  action(input: object, m: TestMessaging) {
-    const out = this.createResponse(input);
+  action(input: AppRequestMessage, m: TestMessaging) {
+    const out = this.createResponse(input as requests);
```

Comment:
Why the `as requests` here? Does `AppRequestMessage` not overlap with the list of possible listener requests?

Reply:
`AppRequestMessage` is indeed a superset, but I typed `createResponse` using `Requests` so it'll error out in the build without the override. I tried to improve on this by making AutomaticResponse a generic using the request types, but afterwards, you can't type a mixed array of them easily as we do here:
https://github.com/finos/FDC3/blob/7b330ac75f627b3c1faddba7f753547b08792b0c/packages/fdc3-agent-proxy/test/support/TestMessaging.ts#L113-L114